modules/metastorage/src/integrationTest/java/org/apache/ignite/internal/metastorage/server/raft/ItMetaStorageRaftGroupTest.java

```diff
@@ -316,22 +326,24 @@ public void onNext(Entry item) {
                                         .stopAsync(componentContext);
                                 assertThat(stopFuture, willCompleteSuccessfully());
 
-                                raftGroupServiceOfLiveServer.refreshLeader().get();
+                                CompletableFuture<LeaderWithTerm> newLeaderWithTermFut = raftGroupServiceOfLiveServer
+                                        .refreshAndGetLeaderWithTerm();
+                                assertThat(newLeaderWithTermFut, willCompleteSuccessfully());
+                                LeaderWithTerm newLeaderWithTerm = newLeaderWithTermFut.join();
 
-                                assertNotSame(oldLeaderId, raftGroupServiceOfLiveServer.leader().consistentId());
+                                assertNotSame(oldLeaderId, newLeaderWithTerm.leader().consistentId());
 
-                                // ensure that leader has been changed only once
-                                assertTrue(
-                                        waitForCondition(() -> replicatorStartedCounter.get() == 4, 5_000),
```

Comment:
Is it possible to drop both replicatorStartedCounter and replcicatorStoppedCounter entirely?
Hope that it is possible to rewrite both onSubscribe and onNext in order to use terms instead of aforementioned counters.

Reply:
As I understand, replicatorStartedCounter is used for waiting for majority in `onSubscribe`. Regarding replicatorStoppedCounter, seems it's not necessary already

Reply:
But since we do wait for refresh leader completeness prior to range() now - explicit majority awaiting logic based on replicaStartedCounter is no longer needer, WDYT?

Reply:
I agree.
Tried 500 repetitions with removed counters: 
https://ci.ignite.apache.org/buildConfiguration/ApacheIgnite3xGradle_Test_IntegrationTests_ModuleMetastorageClient/8389221?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildDeploymentsSection=false&expandBuildChangesSection=true